src/doom/g_game.c

```diff
@@ -851,6 +851,12 @@ void G_DoLoadLevel (void)
     // [crispy] sky texture scales
     R_InitSkyMap();
 
+    // [crispy] pistol start
+    if (crispy->pistolstart && !demorecording && !demoplayback && !netgame)
```

Comment:
There is a shortcut for these checks called `crispy->singleplayer`. Please use this.

Also, I want something to happen if the parameter is given with incompatible boundary conditions. I'd prefer it to error out with a message if given in a network game or during demo recording. For demo playback it depends if the demo is played back from the demo loop or explicitely started from the command line (`singledemo`). In the former case it is alright to silently omit the parameter, but if given together with `-playdemo` or `- timedemo`, the user needs to be informed and the engine quit.

Reply:
Something like this okay for the compatibility check?
```
if ((crispy->pistolstart = M_ParmExists("-pistolstart")) &&
        (M_ParmExists("-record") || M_ParmExists("-playdemo") ||
         M_ParmExists("-timedemo") || netgame))
{
    I_Error("The -pistolstart option is not supported for demos and network play.");
}
```

Reply:
Please no more command line checks at this point in the game. Please check the `crispy->singleplayer` logic.

Reply:
My original thinking was to have it error out during the command line checks in `D_DoomMain()`, prior to the `crispy->singleplayer` being set. Do you have a suggestion for where I should inject the error check? Just do it in `G_DoLoadLevel()`?

Reply:
Yep, I think it's alright to have the check in `G_DoLoadLevel()`. I just don't like the multiple command line checks there, as at this point all the relevant variables are already set. We just need to accept `demoplayback && !singledemo` as a special exception and error out else if `!crispy->singleplayer`.

Reply:
The condition boils down to `!demorecording && (!demoplayback || !singledemo) && !netgame`, right?

Reply:
Take a look at the latest commits. One thing I noticed is that in most cases it won't error out until the first proper level transition. This is due to the first call of `G_DoLoadLevel()` occurring before `demorecording`, `demoplayback` and `netgame` are properly set. Not a big a deal for me, but I wanted to bring it to your attention.

Reply:
Alright, thanks, but it's working as expected now?

Reply:
> The condition boils down to `!demorecording && (!demoplayback || !singledemo) && !netgame`, right?

Yes, you are correct. If the above is `true`, do not call the error and then proceed to either run `G_DoPlayerReborn()` or do nothing. I phrased it a little differently but if you'd like to see this statement in the code I can change things.

Reply:
My phrasing is a bit more compact, but the way you did it is more explicit. Actually, it is written just as I would explain it to a friend. 😉

Reply:
> Alright, thanks, but it's working as expected now?

Looks good on my end.